feat: add user dialog and topbar docs (#3298)
twilio-labs/paste: packages/paste-website/src/pages/components/user-dialog/index.mdx

export const meta = {
  title: 'User Dialog',
  package: '@twilio-paste/user-dialog',

>   description: 'User dialog is an expandable menu that contains profile related actions.',

Reviewer: ```suggestion
  description: 'User Dialog is a menu that contains user profile-related actions.',
```

import {defaultUserDialogExample, imageUserDialogExample} from '../../../component-examples/UserDialogExamples.ts';
export default DefaultLayout;

export const getStaticProps = async () => {
  const navigationData = await getNavigationData();
  const feature = await getFeature('User Dialog');
  return {
    props: {
      data: {
        ...packageJson,
        ...feature,
      },
      navigationData,
    },
  };
};

<NormalizedComponentHeader
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/user-dialog"
  storybookUrl="/?path=/story/components-user-dialog--"
  data={props.data}
/>

---

<contentwrapper>

<PageAside data={mdxHeadings} />

<content>

<LivePreview
  scope={{
    UserDialog,
    UserDialogUserInfo,
    UserDialogUserName,
    UserDialogUserEmail,
    UserDialogContainer,
    UserDialogList,
    UserDialogListItem,
    useUserDialogListState,
    UserDialogSeparator,
    useUID,
    UserIcon,
    ThemeIcon,
    TranslationIcon,
    Box,
    Badge,
  }}
  noInline
  showOverflow
>
  {defaultUserDialogExample}
</LivePreview>

## Guidelines

### About User Dialog

> A User Dialog is a composition of [Avatar](/components/avatar) and [Button](/components/button) that opens a dialog of items that trigger profile-related actions.
> 
> A User Dialog item can be used to perform a single action, such as a page navigation or opening a modal. A profile-related action could be visiting a user settings page, switching the language of a product, or logging out.

Reviewer: ```suggestion
A User Dialog is a composition of [Avatar](/components/avatar), [Button](/components/button), and the [Non-Modal Dialog primitive](/primitives/non-modal-dialog-primitive) that opens a dialog of profile-related actions. This includes visiting a user settings page, switching the language of a product, or logging out.
```
